validator/accounts/v2/wallet/wallet.go

```diff
@@ -100,35 +100,82 @@ func New(cfg *Config) *Wallet {
 	}
 }
 
-// Exists check if a wallet at the specified directory
-// exists and has valid information in it.
-func Exists(walletDir string) error {
+// Exists checks if directory at walletDir exists
+func Exists(walletDir string) (bool, error) {
 	dirExists, err := fileutil.HasDir(walletDir)
 	if err != nil {
-		return errors.Wrap(err, "could not parse wallet directory")
+		return false, errors.Wrap(err, "could not parse wallet directory")
 	}
 	if dirExists {
-		isEmptyWallet, err := isEmptyWallet(walletDir)
-		if err != nil {
-			return errors.Wrap(err, "could not check if wallet has files")
+		return true, nil
+	}
+	return false, nil
+}
+
+// isValidWallet checks if a folder contains a key directory such as `derived`, `remote` or `direct`.
+// Returns true if one of those subdirectories exist, false otherwise.
+func isValidWallet(walletDir string) (bool, error) {
```

Comment:
This is only called by ExistsAndValid.  Separate function to separate the logic.